src/mesido/esdl/esdl_additional_vars_mixin.py

```diff
@@ -16,8 +16,167 @@
+        bounds = self.bounds()
+
+        # ------------------------------------------------------------------------------------------
+        # Limit available pipe classes
+        # TODO: cater for varying temperature when limiting pipe classes below
+        # Here we do a check between the available pipe classes and the demand profiles. This is to
+        # ensure that we don't have unneeded large amount of available pipe classes for pipes
+        # connected to smaller demands.
+        # TODO: add the same for electricity ones we have proper support for that in the ESDLMixin
+        if len(self.temperature_carriers().items()) == 0:
+            for asset, (
+                connected_asset,
+                _orientation,
+            ) in self.energy_system_topology.demands.items():
+                # TODO: add test case for gas once optional gas pipes are used
+                if asset in self.energy_system_components.get("gas_demand", []):
+                    try:
+                        max_demand_g_s = min(
+                            max(self.get_timeseries(f"{asset}.target_gas_demand").values),
+                            bounds[f"{asset}.Gas_demand_mass_flow"][1],
+                        )
+                    except KeyError:
+                        max_demand_g_s = bounds[f"{asset}.Gas_demand_mass_flow"][1]
+                    is_pipe_dn0_needed = True
+                    try:
+                        if min(self.get_timeseries(f"{asset}.target_gas_demand").values) > 0.0:
+                            is_pipe_dn0_needed = False
+                    except KeyError:
+                        is_pipe_dn0_needed = True
+
+                    new_pcs = []
+                    found_pc_large_enough = False
+                    for pc in self.gas_pipe_classes(connected_asset):
+                        if not found_pc_large_enough:
+                            new_pcs.append(pc)
+                            if (
+                                new_pcs[-1].maximum_discharge
+                                * self.parameters(0)[f"{connected_asset}.rho"]
+                                / 1000.00
+                                >= max_demand_g_s
+                            ):  # m3/s * kg/m3 / 1000.0 = g/s
+                                found_pc_large_enough = True
+                    # Remove pipe DN0 from the available pipe list, if there is always flow required
+                    # TODO: Bug to be resolved. Currently the solution is infeasible when only
+                    # 1 pipe class is available, bit it should be able to working.
+                    # Then remove the need for having 2 pipes classes available in new_pcs.
+                    if (
+                        not is_pipe_dn0_needed
+                        and len(new_pcs) > 2
+                        and new_pcs[0].maximum_discharge == 0.0
+                    ):
+                        new_pcs.remove(new_pcs[0])
+
```

Comment:
These lines are the same as 113-118, ideally I would prefer a small internal function before the forloop named "remove_dn0" so that we do not have this TODO twice. And that we call this function to update the new_pcs there and in line 113.